contracts/profit/src/state/idle.rs

```diff
@@ -0,0 +1,153 @@
+use std::marker::PhantomData;
+
+use serde::{Deserialize, Serialize};
+
+use currency::{native::Nls, payment::PaymentGroup};
+use dex::{
+    Account, Enterable, Error as DexError, Handler, Response as DexResponse, Result as DexResult,
+    StartLocalLocalState,
+};
+use finance::{
+    coin::{Coin, CoinDTO, WithCoin, WithCoinResult},
+    currency::{Currency, Group},
+    duration::Duration,
+};
+use platform::{
+    bank::{self, BankAccount, BankAccountView, BankStub, BankView},
+    batch::{Batch, Emitter},
+    message::Response as PlatformResponse,
+    never::{self, Never},
+};
+use sdk::cosmwasm_std::{Deps, Env, QuerierWrapper, Timestamp};
+
+use crate::{msg::ConfigResponse, profit::Profit, result::ContractResult};
+
+use super::{buy_back::BuyBack, Config, ConfigManagement, SetupDexHandler, State, StateEnum};
+
+#[derive(Serialize, Deserialize)]
+pub(super) struct Idle {
+    config: Config,
+    account: Account,
+}
+
+impl Idle {
+    pub fn new(config: Config, account: Account) -> Self {
+        Self { config, account }
+    }
+
+    fn send_nls<B>(&self, env: &Env, account: B) -> ContractResult<Option<(Batch, Emitter)>>
+    where
+        B: BankAccount,
+    {
+        let balance_nls: Coin<Nls> = account.balance()?;
+
+        if balance_nls.is_zero() {
+            Ok(None)
+        } else {
+            Profit::transfer_nls(account, env, self.config.treasury()).map(Some)
+        }
+    }
```

Comment:
It seems unnecessary to send the amount in Nls prior to receiving the exchanged amount past the dex finishes. Suggest removing it.